### PR TITLE
feat(special): make special number form editable

### PR DESCRIPTION
**Pull request by zakariasahmane**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/sva
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| License          | BSD 3-Clause

## Description

Make special number form editable

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html [hunk 1]
```diff
@@ -1,124 +1,142 @@
...
-        data-ng-if="$ctrl.wallet.representative.addressOfResidence.streetType"
-        class="form-group"
-    >
+    <div class="form-group">
         <label
             data-translate="telephony_order_address_street_type"
             for="street-type"
         ></label>
-        <p
+        <input
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html:83]: **nitpick(blocking):** Can you link `label`s and `input`s via `for=""` and `id=""` attributes?

**Review by antleblanc — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import angular from 'angular';
+
+import component from './special-number-form.component';
+
+const moduleName = 'specialNumberForm';
+
+angular
+  .module(moduleName, [])
+  .component('telecomSvaSpecialNumberForm', component)
+  .run(/* @ngTranslationsInject:json ../translations ../identity/translations */);
```
[the remark this answers is not in the document]
Comment (zakariasahmane) [reply packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/index.js:10]: These translations already exist in other components related to SVA, they're just not called in the same place, i think if we do it, we will duplicate too much keys i think.

But it's doable.

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html (hunk 1 above)
Comment (zakariasahmane) [reply packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html:83]: Yes

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            id="cedex"
+            name="cedex"
+            data-ng-model="$ctrl.ngModel.cedex"
+        />
+    </div>
+    <div class="form-group">
+        <label data-translate="telephony_order_address_city" for="city"></label>
+        <input
+            class="oui-input"
+            name="city"
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html:144]:
```suggestion
            name="city"
            id="city"            
```

**Review by antleblanc — CHANGES_REQUESTED** [review]

**Review by antleblanc — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/orderAlias/special/special.controller.js [hunk 4]
```diff
@@ -168,51 +169,49 @@ export default /* @ngInject */ function TelecomTelephonyAliasOrderSpecialCtrl(
...
+      'typology',
+      'country',
+      'city',
+      'adressExtra',
+      'retractation',
+      'displayUniversalDirectory',
+      'pool',
+      'description',
+    ];
+    const form = pick(this.form, fields);
```
Comment (y4nnL) [inline packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/orderAlias/special/special.controller.js:193]: You could use destructuration like so
```javascript
const { amount, numberType, ...form } = this.form
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/orderAlias/special/special.controller.js [hunk 5]
```diff
@@ -168,51 +169,49 @@ export default /* @ngInject */ function TelecomTelephonyAliasOrderSpecialCtrl(
...
+    const companyKind = form.legalform;
     const legalForm = () => {
```
Comment (y4nnL) [inline packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/orderAlias/special/special.controller.js:195]: Why not just use the basic if / else form
```javascript
      if (['UE_UNLISTED_COMPANY', 'UNLISTED_COMPANY'].includes(companyKind)) {
        form.legalform = 'corporation';
      } else if (['CRAFTSMAN', 'CSE', 'EARL', 'FCP', 'FI', 'FOUNDATION'].includes(companyKind)) {
        form.legalform = 'other';
      } else if (companyKind === 'MICRO_ENTERPRISE') {
        form.legalform = 'individual';
      } else {
        form.legalform = 'companyKind.toLowerCase();
      }
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.controller.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { buildEnumList } from '../../sva-wallet.constants';
+
+export default class ReadonlyIdentityController {
```
Comment (y4nnL) [inline packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.controller.js:3]:
```suggestion
export default class SpecialNumberFormController {
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html [hunk 7]
```diff
@@ -0,0 +1,170 @@
+<label
+    for="readonly-wallet-form"
+    data-translate="telephony_billingAccount_svaWallet_kyc_identity_title"
+></label>
+<div name="readonly-wallet-form" class="readonly-wallet-form">
```
Comment (y4nnL) [inline packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/specialNumberForm/special-number-form.html:5]:
```suggestion
<label
    for="special-number-form"
    data-translate="telephony_billingAccount_svaWallet_kyc_identity_title"
></label>
<div name="special-number-form" class="special-number-form">
```

**Review by y4nnL — APPROVED** [review]

**Pull request merged**